app/src/main/java/com/keylesspalace/tusky/components/domainblocks/DomainBlocksRemoteMediator.kt

```diff
@@ -0,0 +1,56 @@
+package com.keylesspalace.tusky.components.domainblocks
+
+import androidx.paging.ExperimentalPagingApi
+import androidx.paging.LoadType
+import androidx.paging.PagingState
+import androidx.paging.RemoteMediator
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.util.HttpHeaderLink
+import retrofit2.HttpException
+import retrofit2.Response
+
+@OptIn(ExperimentalPagingApi::class)
+class DomainBlocksRemoteMediator(
+    private val api: MastodonApi,
+    private val viewModel: DomainBlocksViewModel
```

Comment:
This (passing the viewmodel down to the paging source) is a layering violation. I appreciate there's quite a few of them in the code at the moment, but I'm trying to clear them up in https://github.com/tuskyapp/Tusky/pull/3436.

The approach in that PR is to have the viewmodel create a repository.

The repository manages the cache of items (domain blocks, here), and provides the cache to both the paging source and the remote mediator.

The paging source only reads from the cache. The remote mediator writes to the cache and invalidates (see later comment) the paging source when it does so.